### DISCUSSION TITLE
Why does my Cat Blocks hide when I have it on?

### DISCUSSION THREAD

**Discussion by TheFirstOrangeScratcher7777** [ScratchAddons/ScratchAddons#5860] [Help] [answered]
My cat blocks hide when i have it on, that scary me because i think i bug scartch thats why?

**Comment by Samq64** [comment]
Are you using Scratch Addons version 1.31.1?

**Reply by TheFirstOrangeScratcher7777** [reply]
yes, what's wrong then?

**Reply by Samq64** [reply]
@WorldLanguages What did you set the April Fools timestamp in #5848 to? Are the changes supposed to already be in effect?

**Reply by DNin01** [reply]
> @WorldLanguages What did you set the April Fools timestamp in #5848 to? Are the changes in supposed to already be in effect?

Maybe it's time-zone-sensitive.

**Reply by Samq64** [reply]
> Maybe it's time-zone-sensitive.

No, because it's still the 30th here and cat blocks don't work for me either.

**Reply by DNin01** [reply]
> > Maybe it's time-zone-sensitive.
> 
> 
> 
> No, because it's still the 30th here and cat blocks don't work for me either.

They must've been temporarily disabled regardless of what time it is.

**Reply by WorldLanguages** [reply]
> @WorldLanguages What did you set the April Fools timestamp in #5848 to? Are the changes supposed to already be in effect?

It does not rely on timestamps.
Unfortunately, it looks like the check didn't work as expected (but it makes sense looking back).

**Comment by DNin01** [comment]
My guess is that it was temporarily disabled to avoid conflicting with Scratch's own cat blocks feature, which will become available on April Fools' Day. The cat blocks actually came from Scratch and for a few years now, they have always been accessible on April Fools' Day. But if Scratch Addons tries to turn a cat block into a cat block, things could go wrong.

**Reply by TheFirstOrangeScratcher7777** [reply]
oh, ok, I understand, but today it is not april 1, tomorrow it is

**Reply by DNin01** [reply]
It could either be caused by being in a different time zone than Scratch or Scratch Addons may be disabling the cat blocks ahead of time to prepare for Scratch's update.

**Reply by TheFirstOrangeScratcher7777** [reply]
oh, ok, then, if april 1th are end, that means my cat blocks revive? (i am not american tahts why i write like that)

**Reply by DNin01** [reply]
Scratch Addons' cat blocks will be available again shortly after Scratch finishes celebrating April Fools' Day.

**Reply by TheFirstOrangeScratcher7777** [reply]
ok, i understand, thank you guys very much!

**Comment by WorldLanguages** [comment, marked as answer]
This was not the intended behavior. Addons such as cat blocks, two-column category menu, and others, were supposed to be logically disabled if the Scratch editor was in April Fools mode (which features a "mode" dropdown next to "file" and "edit"). I'm not sure when Scratch will turn on April Fools mode, but that's when it was intended for these addons to stop working, not now.
I don't think it makes sense to try to fix this bug being so close to April 1st anyway.

**Reply by TheFirstOrangeScratcher7777** [reply]
ok, thank you!

**Reply by mxmou** [reply]
> I don't think it makes sense to try to fix this bug being so close to April 1st anyway.

Fixing the bug would also mean that the affected addons would start working again immediately after Scratch disables the April Fools mode.

**Reply by WorldLanguages** [reply]
> Fixing the bug would also mean that the affected addons would start working again immediately after Scratch disables the April Fools mode.

This is a good point

**Reply by WorldLanguages** [reply]
I'm hoping that tomorrow the Scratch Team will ship an update to the Scratch editor, instead of simply toggling the April Fools flag.
If they only toggle the flag, we will need to publish a release to fix this bug.

**Reply by WorldLanguages** [reply]
v1.31.2 was released to fix this
